src/python/zquantum/qcbm/target_thermal_states.py

```diff
@@ -0,0 +1,235 @@
+import bisect
+import typing
+import numpy as np
+from zquantum.core.bitstring_distribution import BitstringDistribution
+from zquantum.core.utils import dec2bin, bin2dec, convert_tuples_to_bitstrings
+
+
+def int2ising(number: int, length: int) -> typing.List[int]:
+    """Converts an integer into a +/-1s bitstring (also called Ising bitstring).
+    Args:
+        number: positive number to be converted into its corresponding Ising bitstring representation
+        length: length of the Ising bitstring (i.e. positive number of spins in the Ising system)
+    Returns:
+        Ising bitstring representation (1D array of +/-1)."""
+
+    binary_bitstring = dec2bin(number, length)
+    ising_bitstring = [bit * 2 - 1 for bit in binary_bitstring]
+    return ising_bitstring
+
+
+def ising2int(ising_bitstring: typing.List[int]) -> int:
+    """Converts a +/-1s bitstring (also called Ising bitstring) into an integer.
+    Args:
+        ising_bitstring: 1D array of +/-1.
+    Returns:
+        Integer number representation of the Ising bitstring."""
+
+    binary_bitstring = [int((bit + 1) / 2) for bit in ising_bitstring]
+    number = bin2dec(binary_bitstring)
+    return number
+
+
+def get_random_hamiltonian_parameters(n_spins: int) -> typing.Tuple:
+    """Generates random h, J, and where h and J are arrays of random coefficients sampled from a normal distribution with zero mean and sqrt(n_spins) sd.
+    For reproducibilty, fix random generator seed in the higher level from which this function is called.
```

Comment:
Asking because it seems like you've thought this through, but why fix the seed at higher levels as opposed to passing it in as a keyword argument?

Reply:
Because if you fix it in here and you generate several random hamiltonians, you will end up having the same exact one. Also in general the seed should be fixed once at the beginning of a script as it's supposed to be a global variable :)